src/DI/DependencyChecker.php

```diff
@@ -70,28 +73,29 @@ public function export(): array
 
 		$classes = array_keys($classes);
 		$functions = array_unique($functions, SORT_REGULAR);
-		$hash = self::calculateHash($classes, $functions);
+		$hash = self::calculateReflectionHash($classes, $functions) . self::calculateCallbackHash($callbacks);
 		$files = @array_map('filemtime', array_combine($files, $files)); // @ - file may not exist
 		$phpFiles = @array_map('filemtime', array_combine($phpFiles, $phpFiles)); // @ - file may not exist
-		return [self::VERSION, $files, $phpFiles, $classes, $functions, $hash];
+		return [self::VERSION, $files, $phpFiles, $classes, $functions, $callbacks, $hash];
```

Comment:
Maybe simpler solution will be better, ie. `$reflectionHash = self::calculateReflectionHash(); $callbackHash = self::calculateCallbackHash();` and then `return [self::VERSION, $files, $phpFiles, $classes, $functions, $callbacks, $reflectionHash, $callbackHash]`, what do you mean?

Reply:
Yes, that's was original solution (linked above). I'll split them, if you wish. How about compatibility with cached container, should I drop it or keep it?

Reply:
Compatibility is not needed, ie. it is enough to have compatible isExpired declaration:

```php
public static function isExpired(int $version, array $files, array &$phpFiles, array $classes, array $functions, string $reflectionHash, array $callbacks = [], string $callbackHash = '')
```